app/AI4Green4Students/Services/PredictionService.cs

```diff
@@ -10,34 +10,35 @@ namespace AI4Green4Students.Services;
 
 public class PredictionService
 {
-  private readonly WorkerOptions _worker;
+  private readonly PredictorOptions _predictor;
 
-  public PredictionService(IOptions<WorkerOptions> worker) => _worker = worker.Value;
+  public PredictionService(IOptions<PredictorOptions> predictor) => _predictor = predictor.Value;
 
   /// <summary>
-  /// Predict products based on reaction smiles.
+  /// Forward prediction based on reaction smiles.
   /// </summary>
   /// <param name="smiles">Reaction smiles.</param>
-  /// <param name="firstN">Number of products to return.</param>
-  /// <returns>Predicted products.</returns>
-  public async Task<List<PredictedProductModel>> PredictProducts(string smiles, int firstN = 5)
+  /// <returns>Results.</returns>
+  public async Task<List<ForwardPredictionModel>> ForwardPrediction(string smiles)
   {
-    var url = _worker.ApiUrl.TrimEnd('/') + "/api"
-      .AppendPathSegment("predict-products")
-      .SetQueryParams(new
-      {
-        smiles, firstN
-      });
+    var url = _predictor.ForwardPredictionApiUrl.TrimEnd('/') + "/api"
+      .AppendPathSegment("predict");
```

Comment:
assuming the `.AppendPathSegment` handles the slashes, couldn't you call it directly on `_predictor.ForwardPredictionApiUrl`?